pandas/io/formats/csvs.py

```diff
@@ -188,6 +198,39 @@ def save(self):
                 for _fh in handles:
                     _fh.close()
 
+    def _index_label_encoder(self):
+        """Encode index label if it is not False.
+
+        Returns:
+            index_label: new index_label given index types
+            encode_labels: list of index labels
+        """
+        index_label = self.index_label
+        obj = self.obj
+        if index_label:
+            # append index label based on index type
+            if isinstance(obj.index, ABCMultiIndex):
+                index_label = []
+                for i, name in enumerate(obj.index.names):
+                    # add empty string is name is None
+                    if name is None:
+                        name = ''
+                    index_label.append(name)
+            else:
+                index_label = obj.index.name
+                # if no name, use empty string
+                if index_label is None:
+                    index_label = ['']
+                else:
+                    index_label = [index_label]
+        elif not isinstance(index_label,
```

Comment:
When does the code go down this branch?

Reply:
sorry, should be `if index_label is True` above. @WillAyd